### ISSUE TITLE
Constructor tear-off working spec does not discuss unnamed and "new" named collision

### ISSUE THREAD

**Issue by srawlins** [dart-lang/language#1645]
(I'm opening a new issue as this is an issue with the current spec; https://github.com/dart-lang/language/issues/1564 is a pretty long thread so I didn't want to rat-hole over there.)

https://github.com/dart-lang/language/blob/master/working/0216%20-%20constructor%20tearoffs/proposal.md

contains:

> This proposal is non-breaking and backwards compatible. Where we introduce new syntactic ambiguities, we retain the current interpretation.

I don't think this is correct. If a class currently has an unnamed constructor and a named constructor named "new", what happens when it upgrades to the constructor tearoff language feature? This presumably will not be allowed with the constructor tearoffs language feature.

The spec does not specify any errors when a class specifies an unnamed constructor _and_ a constructor named `new`.

The answers to these questions may encourage some new compile-time warnings ahead of the language feature, and I'd love to know if we should be implementing these ASAP.

**Comment by leafpetersen** [comment]
> 1. I don't think this is correct. If a class currently has an unnamed constructor and a named constructor named "new

I don't understand, `new` is currently a keyword, and so there can be no existing classes with such a named constructor.

> * The spec does not specify any errors when a class declares an unnamed constructor _and_ a constructor named `new`.
> *

@lrhn this should probably be addressed



> 3\. The spec does not specify any errors when a class declares a static method, getter, setter, or field named `new`

Again, `new` is a keyword, so I believe errors here are specified (cc @eernstg ) and are certainly implemented.  Do you have a case where we are not currently emitting an error?

**Comment by srawlins** [comment]
> I don't understand, new is currently a keyword, and so there can be no existing classes with such a named constructor.

Excellent.

> Again, new is a keyword, so I believe errors here are specified (cc @eernstg ) and are certainly implemented. Do you have a case where we are not currently emitting an error?

Excellent. No I don't have any cases; looks like the correct error is being reported everywhere.

**Comment by lrhn** [comment]
Good point that we should be explicit about unnamed/named-`new` being the same constructor, and it's an error to have two of them.
